realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -720,19 +720,26 @@ public boolean load() {
      */
     @Override
     public boolean contains(Object object) {
-        boolean contains = false;
         if (managedMode) {
             realm.checkIfValid();
+
+            // Deleted objects can never be part of a RealmList
             if (object instanceof RealmObjectProxy) {
                 RealmObjectProxy proxy = (RealmObjectProxy) object;
-                if (proxy.realmGet$proxyState().getRow$realm() != null && realm.getPath().equals(proxy.realmGet$proxyState().getRealm$realm().getPath()) && proxy.realmGet$proxyState().getRow$realm() != InvalidRow.INSTANCE) {
-                    contains = view.contains(proxy.realmGet$proxyState().getRow$realm().getIndex());
+                if (proxy.realmGet$proxyState().getRow$realm() == InvalidRow.INSTANCE) {
```

Comment:
why removing the comparison  of the path ? we need to assure we're comparing managed object from the same Realm?